django/db/migrations/state.py

```diff
@@ -575,25 +575,51 @@ def render(self, apps):
         meta_contents.update(self.options)
         meta = type(str("Meta"), tuple(), meta_contents)
         # Then, work out our bases
-        try:
-            bases = tuple(
-                (apps.get_model(base) if isinstance(base, six.string_types) else base)
-                for base in self.bases
-            )
-        except LookupError:
-            raise InvalidBasesError("Cannot resolve one or more bases from %r" % (self.bases,))
+        bases = list()
+        meta_class_bases = list()
+        meta_class_seen_supertypes = set()
+        for base in self.bases:
+            if isinstance(base, six.string_types):
+                try:
+                    base = apps.get_model(base)
+                except LookupError:
+                    raise InvalidBasesError(
+                        "Cannot resolve one or more bases from %r" % (
+                            self.bases,
+                        )
+                    )
```

Comment:
Now that we know exactly which base lookup failed we could adjust the `InvalidBasesError` message.